test_runner/src/main/kotlin/ftl/adapter/GcStorageDownload.kt

```diff
@@ -0,0 +1,20 @@
+package ftl.adapter
+
+import ftl.adapter.google.toApiModel
+import ftl.api.FileReference
+import ftl.client.google.gcStorageDownload
+import ftl.run.exception.FlankGeneralError
+
+object GcStorageDownload :
+    FileReference.Download,
+    (FileReference, Boolean, Boolean) -> FileReference by { fileReference, ifNeeded, ignoreErrors ->
+        if (!ignoreErrors) {
+            if (fileReference.local.isBlank() && fileReference.remote.isBlank()) throw FlankGeneralError("Cannot create empty FileReference")
+        }
+        if (ifNeeded) {
+            if (fileReference.local.isNotBlank()) fileReference
+            else gcStorageDownload(fileReference.remote, ignoreErrors).toApiModel(fileReference)
+        } else {
+            gcStorageDownload(fileReference.remote, ignoreErrors).toApiModel(fileReference)
+        }
```

Comment:
could you make `when` instead of nested `if`s?

Reply:
👍

Reply:
modified :)